quickstart/src/main/java/com/microsoft/azure/kusto/quickstart/KustoSampleApp.java

```diff
@@ -0,0 +1,465 @@
+        if (!executeControlCommand(kustoClient, databaseName, command)) {
+            System.out.println("Failed to alter the ingestion policy, which could be the result of insufficient permissions. The sample will still run, though ingestion will be delayed for up to 5 minutes.");
+        }
+    }
+
+    private static boolean createIngestionMappings(boolean useExistingMapping, Client kustoClient, String databaseName, String mappingName, String mappingValue, IngestionProperties.DataFormat dataFormat) {
+        if (!useExistingMapping) {
+            if (dataFormat.isMappingRequired() && StringUtils.isBlank(mappingValue)) {
+                System.out.printf("The data format %s requires a mapping, but configuration indicates to not use an existing mapping and no mapping was provided. Skipping this ingestion.%n", dataFormat.name());
+                return false;
+            }
+
+            if (StringUtils.isNotBlank(mappingValue)) {
+                IngestionMapping.IngestionMappingKind ingestionMappingKind = dataFormat.getIngestionMappingKind();
+                waitForUserToProceed(String.format("Create a %s mapping reference named '%s'", ingestionMappingKind, mappingName));
+
+                if (StringUtils.isBlank(mappingName)) {
+                    mappingName = "DefaultQuickstartMapping" + UUID.randomUUID().toString().substring(0, 5);
+                }
+                String mappingCommand = String.format(".create-or-alter table %s ingestion %s mapping '%s' '%s'", tableName, ingestionMappingKind.name().toLowerCase(), mappingName, mappingValue);
+                if (!executeControlCommand(kustoClient, databaseName, mappingCommand)) {
+                    System.out.printf("Failed to create a %s mapping reference named '%s'. Skipping this ingestion.%n", ingestionMappingKind, mappingName);
+                    return false;
+                }
+            }
+        } else if (StringUtils.isBlank(mappingName)) {
+            System.out.println("The configuration indicates an existing mapping should be used, but none was provided. Skipping this ingestion.");
+            return false;
+        }
+        return true;
+    }
+
+    private static void ingestData(Map<String, String> file, IngestionProperties.DataFormat dataFormat, IngestClient ingestClient, String databaseName, String tableName, String mappingName) {
+        String sourceType = file.get("sourceType");
+        String uri = file.get("dataSourceUri");
+        waitForUserToProceed(String.format("Ingest '%s' from %s", uri, sourceType));
+        // Tip: When ingesting json files, if each line represents a single-line json, use MULTIJSON format even if the file only contains one line.
+        //  If the json contains whitespace formatting, use SINGLEJSON. In this case, only one data row json object is allowed per file.
+        if (dataFormat == IngestionProperties.DataFormat.json) {
+            dataFormat = IngestionProperties.DataFormat.multijson;

```

Comment:
That's a Quickstart-specific design decision, so that they're less likely to fail on this common issue.
The comment is there to clarify why it's being done.